dash/include/dash/algorithm/Copy.h

```diff
@@ -105,108 +122,81 @@ template <
   typename ValueType,
   class GlobInputIt >
 ValueType * copy_impl(
-  GlobInputIt                  in_first,
-  GlobInputIt                  in_last,
+  GlobInputIt                  begin,
+  GlobInputIt                  end,
   ValueType                  * out_first,
   std::vector<dart_handle_t> & handles)
 {
-  DASH_LOG_TRACE("dash::copy_impl()",
-                 "in_first:",  in_first.pos(),
-                 "in_last:",   in_last.pos(),
+  DASH_LOG_TRACE("dash::copy_impl() global -> local",
+                 "in_first:",  begin.pos(),
+                 "in_last:",   end.pos(),
                  "out_first:", out_first);
-  auto pattern = in_first.pattern();
-  typedef typename decltype(pattern)::index_type index_type;
-  typedef typename decltype(pattern)::size_type  size_type;
-  size_type num_elem_total = dash::distance(in_first, in_last);
+  typedef typename GlobInputIt::pattern_type::size_type  size_type;
```

Comment:
still using `GlobInputIt::pattern_type::size_type` instead of just 'GlobInputIt::size_type`